packages/amplify-graphql-relational-transformer/src/resolver/rds-generator.ts

```diff
@@ -149,10 +153,41 @@ export class RDSRelationalResolverGenerator extends RelationalResolverGenerator
   /**
    * Generate connection response template for RDS.
    */
-  generateConnectionLambdaResponseMappingTemplate = (): string => {
+  generateSingleItemConnectionLambdaResponseMappingTemplate = (): string => {
     const statements: Expression[] = [];
     statements.push(
-      ifElse(ref('ctx.error'), methodCall(ref('util.error'), ref('ctx.error.message'), ref('ctx.error.type')), toJson(ref('ctx.result'))),
+      ifElse(
+        ref('ctx.error'),
+        methodCall(ref('util.error'), ref('ctx.error.message'), ref('ctx.error.type')),
+
+        // Make sure the retrieved item has the __operation field, so the individual type resolver can appropriately redact fields
+        compoundExpression([
+          set(ref('resultValue'), ref('ctx.result')),
+          set(ref('operation'), methodCall(ref('util.defaultIfNull'), methodCall(ref('ctx.source.get'), str(OPERATION_KEY)), nul())),
+          iff(equals(ref('operation'), str('Mutation')), qref(methodCall(ref('resultValue.put'), str(OPERATION_KEY), str('Mutation')))),
+          toJson(ref('resultValue')),
+        ]),
+      ),
+    );
+    return printBlock('ResponseTemplate')(compoundExpression(statements));
+  };
+
+  generateListConnectionLambdaResponseMappingTemplate = (): string => {
+    const statements: Expression[] = [];
+    statements.push(
+      ifElse(
+        ref('ctx.error'),
+        methodCall(ref('util.error'), ref('ctx.error.message'), ref('ctx.error.type')),
+        // Make sure each retrieved item has the __operation field, so the individual type resolver can appropriately redact fields
+        compoundExpression([
+          set(ref('resultValue'), ref('ctx.result')),
```

Comment:
here we're assigning the `ctx.result` to a new value, mutating the new value and returning it.
Functionally I don't think it makes a difference, but why have 2 approaches? Treat this as a nit.

Reply:
I wondered that myself, and elected to leave it because it was closer to the style of the surrounding code. When we merge this fix into the `feature/gen2-release` branch we can look at normalizing this.